### PR TITLE
feat: add co sign button initiating file upload drawer

### PR DESCRIPTION
**Pull request by szymonmaslowski**
# Checklist

- [x] JIRA - [\<link>](https://input-output.atlassian.net/browse/LW-10777)
- [ ] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

- added co sign button
  - I had trouble with the icon, so I skipped it. I need to create a ticket for that
- a first step of the co-sign flow with file import UI
  - not functional yet. You can select a file but it does not get validated and the continue button does not work. Validation, creating a transaction and activating the continue button will go in next PR(s)
  - the header of the drawer is incorrect. It says `Send` but should be `Co sign transaction`. TBD.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

image
image
image

### CODE REVIEW

File: apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/components/SendTransactionLayout.tsx [hunk 1]
```diff
@@ -60,37 +39,6 @@ export const SendTransactionLayout = withAddressBookContext(
       }
     }, [closeDrawer, currentProvider, blockchainProvider, setCurrentProvider]);
 
-    const shouldAssetPickerDisplayFooter = multipleSelectionAvailable && currentSection === Sections.ASSET_PICKER;
```
Comment (szymonmaslowski) [inline apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/components/SendTransactionLayout.tsx:63]: I moved it to a hook reused from two flows: send and co sign

File: apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/useOpenTransactionDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/* eslint-disable react/no-multi-comp */
```
Comment (szymonmaslowski) [inline apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/useOpenTransactionDrawer.tsx:1]: Render functions are seen as separate components

File: apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/useOpenTransactionDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  } = useTransactionSections();
+  const [multipleSelectionAvailable] = useMultipleSelection();
+  const shouldAssetPickerDisplayFooter = multipleSelectionAvailable && currentSection === Sections.ASSET_PICKER;
+
+  const shouldRenderFooter =
+    ![
+      Sections.ADDRESS_LIST,
+      Sections.ADDRESS_FORM,
+      Sections.ASSET_PICKER,
+      Sections.IMPORT_SHARED_WALLET_TRANSACTION_JSON
```
Comment (szymonmaslowski) [inline apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/useOpenTransactionDrawer.tsx:39]: If it is a `IMPORT_SHARED_WALLET_TRANSACTION_JSON` step then footer would not be rendered. Instead, the `CoSignEntry` component renders the foot buttons on its own.

File: packages/core/src/ui/components/TransactionCTAs/TransactionCTAs.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      <Button className={buttonClassName} block onClick={onReceiveClick} color="gradient" data-testid="send-button">
+        <ArrowDiagonalUp className={classnames(styles.icon, !popupView && styles.iconInExpandedView)} />
+        {t('core.transactionCtas.receive')}
+      </Button>
+      <Button className={buttonClassName} block onClick={onSendClick} color="gradient" data-testid="receive-button">
+        <ArrowDiagonalDown className={classnames(styles.icon, !popupView && styles.iconInExpandedView)} />
+        {t('core.transactionCtas.send')}
+      </Button>
+      {onCoSignClick && (
+        <Button className={buttonClassName} block onClick={onCoSignClick} color="gradient" data-testid="co-sign-button">
```
Comment (szymonmaslowski) [inline packages/core/src/ui/components/TransactionCTAs/TransactionCTAs.tsx:36]: TODO: add icon. I would do it in separate ticket

**Comment by pczeglik-iohk** [comment]
# Allure Report
[`allure-report-publisher`](https://github.com/andrcuns/allure-report-publisher) generated test report!



**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/smoke/linux/chrome/6370/10093509747/index.html) for [73297bcc](https://github.com/input-output-hk/lace/pull/1322/commits/73297bcc317db4f4020d11fc810414e6591f6995)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 25     | 7      | 0       | 0     | 32    | ❌     |

File: apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/components/Transaction.tsx [hunk 5]
```diff
@@ -57,9 +59,13 @@ export const SendTransaction = withAddressBookContext(
...
 
-    // TODO: move isPopupView to store, jira ticket need to be added https://input-output.atlassian.net/browse/LW-5296
     const sectionMap: Record<Sections, React.ReactElement> = {
-      [Sections.FORM]: <TransactionForm isPopupView={isPopupView} />,
+      ...(config.content === DrawerContent.SEND_TRANSACTION && {
+        [Sections.FORM]: <TransactionForm isPopupView={isPopupView} />
+      }),
+      ...(config.content === DrawerContent.CO_SIGN_TRANSACTION && {
+        [Sections.IMPORT_SHARED_WALLET_TRANSACTION_JSON]: <ImportSharedWalletTransaction />
+      }),
```
Comment (vetalcore) [inline apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/components/Transaction.tsx:68]:
```suggestion
        [Sections.FORM]: <TransactionForm isPopupView={isPopupView} />,
        [Sections.IMPORT_SHARED_WALLET_TRANSACTION_JSON]: <ImportSharedWalletTransaction />,
```

File: packages/core/src/shared-wallets/transaction/CoSignEntry.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  };
+
+  const handleContinue = () => {
+    onContinue();
+  };
+
+  return (
+    <Flex gap="$32" flexDirection="column" h="$fill">
+      <Flex gap="$8" flexDirection="column">
+        <Text.SubHeading>Import transaction</Text.SubHeading>
```
Comment (vetalcore) [inline packages/core/src/shared-wallets/transaction/CoSignEntry.tsx:64]: please move strings to translation package

**Review by vetalcore — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/src/views/browser-view/components/Drawer/DrawerUIContent.tsx [hunk 7]
```diff
@@ -1,14 +1,15 @@
...
 import { useDrawer } from '../../stores';
 import { Drawer } from '@lace/common';
 import { QRInfoWalletDrawer } from '../QRInfoWalletDrawer';
 import { WalletUsedAddressesDrawer } from '../WalletUsedAddressesDrawer';
 import { Skeleton } from 'antd';
 
 export enum DrawerContent {
-  SEND_TRANSACTION = 'send-trasaction',
+  SEND_TRANSACTION = 'send-transaction',
+  CO_SIGN_TRANSACTION = 'co-sign-transaction',
```
Comment (shawnbusuttil) [inline apps/browser-extension-wallet/src/views/browser-view/components/Drawer/DrawerUIContent.tsx:12]: Small: `co-sign` get rid of the hyphen here.

File: apps/browser-extension-wallet/src/views/browser-view/components/Drawer/DrawerUIContent.tsx [hunk 8]
```diff
@@ -26,7 +27,9 @@ export interface DrawerConfig {
 const renderDrawerContent = (content?: DrawerContent) => {
   switch (content) {
     case DrawerContent.SEND_TRANSACTION:
-      return <SendTransaction />;
+      return <Transaction />;
+    case DrawerContent.CO_SIGN_TRANSACTION:
+      return <Transaction />;
```
Comment (shawnbusuttil) [inline apps/browser-extension-wallet/src/views/browser-view/components/Drawer/DrawerUIContent.tsx:32]: Can you stack the cases here?

File: packages/core/src/shared-wallets/transaction/CoSignEntry/CoSignEntry.tsx [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+      }
+      try {
+        resolve(JSON.parse(reader.result as string));
+      } catch {
+        reject(new UnrecognizedFile());
+      }
+    });
+    reader.readAsText(file);
+  });
+
```
Comment (shawnbusuttil) [inline packages/core/src/shared-wallets/transaction/CoSignEntry/CoSignEntry.tsx:29]: Can you extract the error and util function out here?

**Review by shawnbusuttil — CHANGES_REQUESTED** [review]
Left some suggestions.

**Review by shawnbusuttil — APPROVED** [review]

**Review by vetalcore — APPROVED** [review]

**Pull request merged**